### PR TITLE
Feature/all other evidence ref

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**
For https://dev.azure.com/civica-digital/EA%20-%20WEEE/_workitems/edit/43364. 

Also refactors some of the list code to try and make better reuse of code.

### CODE REVIEW

File: src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/Overview/_SearchByReferencePartial.cshtml [hunk 1]
```diff
@@ -2,13 +2,15 @@
...
-            @Html.Gds().TextBoxFor(m => m.FilterViewModel.SearchRef)
-            <button class="govuk-button"><i class="fa fa-search" aria-hidden="true"></i><span class="sr-only govuk-visually-hidden">Search by reference number</span></button>
+    @Html.Hidden("activeOverviewDisplayOption", ViewData["activeOverviewDisplayOption"])
+
+    <div class="govuk-!-width-full govuk-panel-grey-background">
+        <div class="govuk-form-group govuk-!-width-one-half govuk-!-margin-left-2 govuk-!-padding-top-2">
+            <div class="govuk-form-group @Html.Gds().FormGroupClass(m => m.FilterViewModel.SearchRef)">
+                @Html.Gds().LabelFor(m => m.FilterViewModel.SearchRef, new { @class = "govuk-label--s" }, false)
+                @Html.Gds().TextBoxFor(m => m.FilterViewModel.SearchRef)
+                <button class="govuk-button"><i class="fa fa-search" aria-hidden="true" title="search by reference"></i><span class="sr-only">Search by reference number</span></button>
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/Overview/_SearchByReferencePartial.cshtml:12]: just wondering if there is a Gds control for it?

File: src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/Overview/ViewAllOtherEvidenceOverview.cshtml [hunk 2]
```diff
@@ -43,56 +47,58 @@
...
-                                {
-                                    <td class="govuk-table__cell">@note.SubmittedDate.Value.ToShortDateString()</td>
-                                }
-                                else
-                                {
-                                    <td class="govuk-table__cell"></td>
-                                }
-                                <td class="govuk-table__cell">@note.Recipient</td>
-                                @if (note.TypeOfWaste.HasValue)
+                                foreach (EvidenceNoteRowViewModel note in Model.EvidenceNotesDataList)
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/Overview/ViewAllOtherEvidenceOverview.cshtml:52]: could be a var

**Review by meglenapnovoselska — APPROVED** [review]
I'm happy with the changes

File: src/EA.Weee.Core.Tests.Unit/Helpers/NoteTypeCoreData.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    public class NoteTypeCoreData : IEnumerable<object[]>
+    {
+        public IEnumerator<object[]> GetEnumerator()
+        {
+            foreach (var status in Enum.GetValues(typeof(EA.Weee.Core.AatfEvidence.NoteType)))
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Core.Tests.Unit/Helpers/NoteTypeCoreData.cs:11]: please change to type

File: src/EA.Weee.Web.Tests.Unit/Areas/Aatf/Mapping/ToViewModel/AllOtherNotesViewModelMapTests.cs [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            var model = new EditDraftReturnedNotesViewModel
+            {
+                EvidenceNotesDataList = returnedNotes
+            };
+
+            var organisationId = Guid.NewGuid();
+            var aatfId = Guid.NewGuid();
+
+            var transfer = new EvidenceNotesViewModelTransfer(organisationId, aatfId, notes);
+            A.CallTo(() => mapper.Map<List<EvidenceNoteRowViewModel>>(notes)).Returns(returnedNotes);
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web.Tests.Unit/Areas/Aatf/Mapping/ToViewModel/AllOtherNotesViewModelMapTests.cs:130]: would you add it to the assert area - A.CallTo(() => mapper.Map<List<EvidenceNoteRowViewModel>>(notes)).MustHaveHappenedExactylOnce()

File: src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/Overview/EditDraftReturnedNotesOverview.cshtml [hunk 5]
```diff
@@ -42,9 +44,9 @@
                             </tr>
                         </thead>
                         <tbody class="govuk-table__body">
-                            @if (Model.ListOfNotes.Any())
+                            @if (Model.EvidenceNotesDataList.Any())
                             {
-                                foreach (EditDraftReturnedNote note in Model.ListOfNotes)
+                                foreach (EvidenceNoteRowViewModel note in Model.EvidenceNotesDataList)
```
Comment (meglenapnovoselska) [inline src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/Overview/EditDraftReturnedNotesOverview.cshtml:49]: var

File: src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/Overview/_SearchByReferencePartial.cshtml (hunk 1 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/Overview/_SearchByReferencePartial.cshtml:12]: Yeah will look at moving this into a control when we add the search to the other PCS screen

File: src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/Overview/EditDraftReturnedNotesOverview.cshtml (hunk 5 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/Overview/EditDraftReturnedNotesOverview.cshtml:49]: updated

File: src/EA.Weee.Core.Tests.Unit/Helpers/NoteTypeCoreData.cs (hunk 3 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Core.Tests.Unit/Helpers/NoteTypeCoreData.cs]: updated

File: src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/Overview/ViewAllOtherEvidenceOverview.cshtml (hunk 2 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Aatf/Views/ManageEvidenceNotes/Overview/ViewAllOtherEvidenceOverview.cshtml]: updated

**Pull request merged**